### PR TITLE
Add debtor account currency validation

### PR DESCRIPTION
**Pull request by isharailanga**
This PR is to fix https://github.com/wso2-enterprise/financial-open-banking/issues/7373.

### CODE REVIEW

File: en/docs/learn/integration-with-bank.md [hunk 1]
```diff
@@ -229,6 +229,19 @@ authorisation scenario in the Payments API:
  3. Each owner should perform the authorisation of the consent.
  4. Upon consent approval of all owners, the consent status changes to `ACCP`.
  
+#### Debtor Account Currency Validation
+
+During the authorization process, you can perform a currency validation for the debtor account.
+
+If the payment is made from a sub-account when there is not enough balance in the original debtor account:
+
+- The bank should implement a custom authorization retrieval step to check for other sub-accounts for the required
```
Comment (Akila94) [inline en/docs/learn/integration-with-bank.md]: We can link the documentation about [the retrieval step extension](https://ob.docs.wso2.com/en/latest/develop/consent-management-authorize/#retrieve) here.

File: en/docs/learn/integration-with-bank.md [hunk 2]
```diff
@@ -229,6 +229,19 @@ authorisation scenario in the Payments API:
...
+  balances to make the payment.
+- If the payment is made from a different debtor account and not the one provided with the consent, the bank should 
+  customise the UI to display the account from which the payment amount was deducted.
+- The bank needs to implement a custom authorization persist step to amend the original consent with the account
```
Comment (Akila94) [inline en/docs/learn/integration-with-bank.md]: Can link [the persist steps extension doc](https://ob.docs.wso2.com/en/latest/develop/consent-management-authorize/#persist) here.

File: en/docs/learn/integration-with-bank.md [hunk 3]
```diff
@@ -229,6 +229,19 @@ authorisation scenario in the Payments API:
...
  
+#### Debtor Account Currency Validation
+
+During the authorization process, you can perform a currency validation for the debtor account.
+
+If the payment is made from a sub-account when there is not enough balance in the original debtor account:
+
+- The bank should implement a custom authorization retrieval step to check for other sub-accounts for the required
+  balances to make the payment.
+- If the payment is made from a different debtor account and not the one provided with the consent, the bank should
```
Comment (Akila94) [inline en/docs/learn/integration-with-bank.md:240]: Can link [the consent page customization doc](https://ob.docs.wso2.com/en/latest/develop/customize-consent-page/) here.

File: en/docs/learn/integration-with-bank.md (hunk 1 above)
Comment (isharailanga) [reply en/docs/learn/integration-with-bank.md]: Fixed in 2ddda5b6f70220be916e949f330b816fc8aa54fa.

File: en/docs/learn/integration-with-bank.md (hunk 2 above)
Comment (isharailanga) [reply en/docs/learn/integration-with-bank.md]: Fixed in 2ddda5b6f70220be916e949f330b816fc8aa54fa

File: en/docs/learn/integration-with-bank.md (hunk 3 above)
Comment (isharailanga) [reply en/docs/learn/integration-with-bank.md:240]: Fixed in 2ddda5b6f70220be916e949f330b816fc8aa54fa

**Review by Akila94 — APPROVED** [review]

**Pull request merged**